src/WebServerManager.cpp

```
@@ -363,84 +363,64 @@ void WebServerManager::handleApiRegisterUid() {
-    code = http.POST(spoolJson);
-    if (code != 200 && code != 201) {
+    if (spoolId > 0) {
+        // Spool exists — update it instead of creating a duplicate
+        Serial.printf("register-uid: found existing spool %d, updating\n", spoolId);
+        float effInitial = initialWeight > 0 ? initialWeight : existingInitialWeight;
+        if (!enrichUpdateSpool(client, http, baseUrl, spoolId, filamentId, remainingWeight, effInitial)) {
+            xSemaphoreGive(g_httpMutex);
+            sendError(500, "Failed to update existing spool");
+            return;
+        }
+    } else {
```

Clang (14.0.6)

[note] +1, nesting level increased to 1

(clang)